go/mysql/server.go

```diff
@@ -335,60 +336,30 @@ func (l *Listener) handle(conn net.Conn, connectionID uint32, acceptTime time.Ti
 			queryStart := time.Now()
 			query := c.parseComQuery(data)
 			c.recycleReadPacket()
-			fieldSent := false
-			// sendFinished is set if the response should just be an OK packet.
-			sendFinished := false
-			err := l.handler.ComQuery(c, query, func(qr *sqltypes.Result) error {
-				if sendFinished {
-					// Failsafe: Unreachable if server is well-behaved.
-					return io.EOF
-				}
-
-				if !fieldSent {
-					fieldSent = true
-
-					if len(qr.Fields) == 0 {
-						sendFinished = true
-						// We should not send any more packets after this.
-						return c.writeOKPacket(qr.RowsAffected, qr.InsertID, c.StatusFlags, 0)
-					}
-					if err := c.writeFields(qr); err != nil {
-						return err
+			queries := []string{query}
```

Comment:
This could be:
var queries []string
if ... {
  ...
  queries = stmts
} else {
  queries = []string{query}
}

to avoid an allocation in one case.

Reply:
done